### PR TITLE
[Core][Workflow] Set default storage if there is no storage specified when starting a local cluster

### LINKED ISSUE #27046: [workflow] We should give the storage a default value if it's not set in some way.
### What happened + What you expected to happen

```
import ray
import ray.workflow as workflow
ray.init()
workflow.init()


@ray.remote
def f(a):
    print(a)

@ray.remote
def g():
    return 10


workflow.run(f.bind({"X": g.bind()}), workflow_id="simple")
```

Will throw 
```
RuntimeError: No storage URI has been configured for the cluster. Specify a storage URI via `ray.init(storage=<uri>)`
```

This is bad for local testing. We should use a default one instead.

### Versions / Dependencies

master

### Reproduction script

```
import ray
import ray.workflow as workflow
ray.init()
workflow.init()


@ray.remote
def f(a):
    print(a)

@ray.remote
def g():
    return 10


workflow.run(f.bind({"X": g.bind()}), workflow_id="simple")
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by suquark**
Signed-off-by: Siyuan Zhuang <suquark@gmail.com>

## Why are these changes needed?

Currently if users call `ray.init()` to start a ray cluster locally without specifying `storage`, then all libraries that depends on the storage could not work because this would happen when trying to access the storage:

```python
RuntimeError: No storage URI has been configured for the cluster. Specify a storage URI via `ray.init(storage=<uri>)`
```

## Related issue number

Closes https://github.com/ray-project/ray/issues/27046

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/worker.py [hunk 1]
```diff
@@ -1355,6 +1355,14 @@ def init(
         else:
             usage_lib.set_usage_stats_enabled_via_env_var(False)
 
+        # assign a default storage if there is no storage.
```
Comment (iycheng) [inline python/ray/_private/worker.py:1358]: I'm wondering whether we also need to print any logs for this.

**Review by ericl — COMMENTED** [review]
This only applies for ray.init() right, not ray start? We can't set storage for multi-node clusters by default safely.

File: python/ray/_private/worker.py (hunk 1 above)
Comment (ericl) [reply python/ray/_private/worker.py:1358]: We can't print any log spam on ray init.

**Comment by suquark** [comment]
@ericl yes, it only sets a default storage when `ray.init()` locally without connecting to a cluster.

**Review by ericl — APPROVED** [review]

**Comment by suquark** [comment]
a few CI failures maybe related

**Pull request closed without merging**